### ISSUE TITLE
Login/create account -> AccountView UI blocking or extra feedback required

### ISSUE THREAD

**Issue by mattsmithies** [blockmason/lndr-mobile#17]
Briefly, when logging in or creating the account

Seems really slow, is it the transition from initial -> AccountView or is the await from confirmAccount/loginAccount (async) taking the time?

Let me know if you get this issue?

**Comment by NateFerrero** [comment]
Hmm did you have remote debugging enabled? Sometimes that can pause the JS engine. Will look into this.

**Comment by mattsmithies** [comment]
It looks like the perf issue on Android is related to these lines:

const seed = mnemonicInstance.toSeed(PASSWORD_SALT + password)
const privateKey = mnemonicInstance.toHDPrivateKey(password)

Therefore the bitcore-mnemonic lib could be the issue

**Comment by mattsmithies** [comment]
Otherwise could we just future up this functionality, while transitioning to the account screen to make it more UX friendly.

**Comment by mattsmithies** [comment]
Looks like the issue is probably within the SecureRandom implementation inside of Android as it blocks when entropy is gathered from /dev/random

**Comment by aupiff** [comment]
Yeah, I've seen this before, /dev/urandom is what people use to avoid blocking.

https://sockpuppet.org/blog/2014/02/25/safely-generate-random-numbers/

**Comment by aupiff** [comment]
This https://tersesystems.com/blog/2015/12/17/the-right-way-to-use-securerandom/ says SecureRandom will use /dev/urandom if instantiated thus:

```java
SecureRandom nativeRandom = SecureRandom.getInstance("NativePRNGNonBlocking"); // assuming Unix
```

**Comment by mattsmithies** [comment]
Thanks! Been looking at something similar. Admittedly a little concerned after reading this. http://www.infosecisland.com/blogview/24773-Android-N-Deprecating-Crypto-Provider-and-SHA1PRNG-Algorithm.html

**Comment by aupiff** [comment]
Yeah, Android sucks. Many wallet hacks on Android have happened in the recent past due to their standard PRNGs.

**Comment by mattsmithies** [comment]
Also according to this the default usage of SecureRandom seems to use dev/urandom. https://tersesystems.com/blog/2015/12/17/the-right-way-to-use-securerandom/

**Comment by aupiff** [comment]
You're right. Are you blocking while using /dev/urandom, then?

**Comment by mattsmithies** [comment]
Resolves in PR at #47
